homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -193,17 +219,7 @@ def extra_state_attributes(self):
         return attributes
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def icon(self):
-        """Return the icon for the frontend."""
-        return ICON
-
-    @property
-    def native_value(self):
+    def native_value(self) -> datetime | None:
```

Comment:
Could you go one step further and simply set `self._attr_native_value` inside `async_update`?